x-pack/plugins/ml/public/components/kql_filter_bar/__tests__/utils.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+
+import expect from 'expect.js';
+import { removeFilterFromQueryString } from '../utils';
+
+describe('ML - KqlFilterBar utils', () => {
+
+  describe('removeFilterFromQueryString', () => {
+
+    const fieldName = 'http.response.status_code';
+    const fieldValue = '200';
+
+    it('removes selected fieldName/fieldValue from query string with one value', () => {
+      const currentQueryString = 'http.response.status_code : "200"';
+      const expectedOutput = '';
+      const result = removeFilterFromQueryString(currentQueryString, fieldName, fieldValue);
+      expect(result).to.be(expectedOutput);
+    });
+
+    it('removes selected fieldName/fieldValue of type number from existing query string with one value', () => {
+      const currentQueryString = 'http.response.status_code : 200';
+      const expectedOutput = '';
+      const result = removeFilterFromQueryString(currentQueryString, fieldName, fieldValue);
+      expect(result).to.be(expectedOutput);
+    });
+
+    it('removes selected fieldName/fieldValue from query string with multiple values', () => {
```

Comment:
Is it worth changing this test so that it removes the fieldName/value with a String type value. Otherwise it is basically the same as the test below? Or edit the descriptions to reflect that you are testing removing the first / second clause in the filter.

Reply:
I'm not sure what you mean here. This test ^ is checking that the value "200" (wrapped with double quotes) gets removed while the one below tests that the removal also works when the value is not a string/not wrapped with double quotes. First and second clause are also valuable tests but I thought putting that in the description would make it too long.

Reply:
Ah ok, I missed the quotes around the `200` had done from the second test. LGTM!